docs/get-started/getting-started-kubernetes.md

```diff
@@ -0,0 +1,256 @@
+     --conf spark.rapids.memory.pinnedPool.size=2G \
+     --conf spark.locality.wait=0s \
+     --conf spark.sql.files.maxPartitionBytes=512m \
+     --conf spark.sql.shuffle.partitions=10 \
+     --conf spark.plugins=com.nvidia.spark.SQLPlugin \
+     --conf spark.kubernetes.namespace=$SPARK_NAMESPACE  \
+     --conf spark.executor.resource.gpu.discoveryScript=/opt/sparkRapidsPlugin/getGpusResources.sh \
+     --conf spark.executor.resource.gpu.vendor=nvidia.com \
+     --conf spark.kubernetes.container.image=$IMAGE_NAME \
+     --conf spark.executor.extraClassPath=/opt/sparkRapidsPlugin/rapids-4-spark_<version>.jar:/opt/sparkRapidsPlugin/cudf-<version>.jar   \
+     --driver-class-path=./cudf-<version>.jar:./rapids-4-spark_<version>.jar \
+     --driver-memory 2G 
+```
+
+Only `client` mode is applicable. If you specify this as `cluster` mode, you would see below error:
+```shell
+Cluster deploy mode is not applicable to Spark shells.
+```
+And also you notice that here we removed `--conf spark.driver.extraClassPath` but added `--driver-class-path`.
+This is because now the Driver is running on the client machine so the jar paths should be local machine path.
+
+Only the Executor POD(s) are running inside Kubernetes:
+```
+$ kubectl get pods
+NAME                                     READY   STATUS    RESTARTS   AGE
+mysparkshell-bfe52e782f44841c-exec-1     1/1     Running   0          11s
+```
+
+We can run below scala code in `spark-shell` to test if GPU is enabled for our queries:
+```shell
+val df = spark.sparkContext.parallelize(Seq(1)).toDF()
+df.createOrReplaceTempView("df")
+spark.sql("SELECT value FROM df WHERE value <>1").show
+spark.sql("SELECT value FROM df WHERE value <>1").explain
+:quit
+```
+The expected explain plan should contain the GPU related operators:
+```shell
+scala> spark.sql("SELECT value FROM df WHERE value <>1").explain
+== Physical Plan ==
+GpuColumnarToRow false
++- GpuFilter NOT (value#2 = 1)
+   +- GpuRowToColumnar TargetSize(2147483647)
+      +- *(1) SerializeFromObject [input[0, int, false] AS value#2]
+         +- Scan[obj#1]
+```
+
+### spark-submit python code in `client` mode.
+
+Of course, you can `COPY` the python code in the Docker image when building it.
+Then you can spark-submit it using `cluster` mode as previous example pi job.
+
+However if you do not want to re-build the Docker image each time and just want to submit the python code
+from the client machine, you can use `client` mode.
+
+```shell
+$SPARK_HOME/bin/spark-submit \
+     --master $K8SMASTER \
+     --deploy-mode client  \
+     --name mypythonjob \
+     --conf spark.rapids.sql.concurrentGpuTasks=1 \
+     --conf spark.executor.instances=1 \
+     --conf spark.executor.resource.gpu.amount=1 \
+     --conf spark.executor.memory=4G \
+     --conf spark.executor.cores=1 \
+     --conf spark.task.cpus=1 \
+     --conf spark.task.resource.gpu.amount=1 \
+     --conf spark.rapids.memory.pinnedPool.size=2G \
```

Comment:
Need to set `spark.kubernetes.memoryOverheadFactor` as well.

Reply:
@jlowe Do you mean the default spark.kubernetes.memoryOverheadFactor=0.1 is not enough? What is your suggestion on the value of spark.kubernetes.memoryOverheadFactor is we pin 2G memory in this case?

Reply:
No, 0.1 is not going to be enough even in this trivial example because 4G heap * 0.1 overhead is only ~400M of overhead, and that's not enough to cover the 2G of pinned memory (along with regular JVM overhead).  In this case it would need to be set to something like 0.6 with 0.5 to cover the 2G of pinned memory and the remaining 0.1 to cover the JVM itself as for normal Spark-on-k8s JVM jobs.

Reply:
@jlowe i just added spark.kubernetes.memoryOverheadFactor=0.6 can you take a look?thx 
I tested this in example job and it works.

Reply:
personally I don't like recommending changing the overhead factor as it changes with the amount of memory you specify, I would prefer to see us use 

spark.executor.memoryOverhead

which is not k8s specific so it applies to multiple resource managers and you set it at a value. If we know pinned memory is going to be 2G, set it to 3G for instance. This lets you change the executor memory size without having to worry about changing this one as well.

Reply:
Yes, +1.  I should have pointed to spark.executor.memoryOverhead.

Reply:
Yep just Changed to spark.executor.memoryOverhead=3G